test/testutil.c

```diff
@@ -145,19 +145,110 @@ int run_tests(const char *test_prog_name)
     return EXIT_SUCCESS;
 }
 
+void fail_message(const char *desc, va_list args, const char *type,
```

Comment:
It would be possible to have macro versions of these without the description or at least without the description with arguments.  That might be sufficient.  I'll mulled it over a bit....